pkg/agent/protocol/protocol.go

```diff
@@ -29,9 +29,21 @@ type Disruptor interface {
 type Proxy interface {
 	Start() error
 	Stop() error
+	// Metrics returns a map of counter-type metrics. Implementations may return zero or more of the metrics defined
+	// below, as well as any number of implementation-defined metrics.
+	Metrics() map[string]uint
 	Force() error
 }
 
+const (
+	// MetricRequests is the total number of requests received by the proxy.
+	MetricRequests = "requests_total"
+	// MetricRequestsExcluded is the total number of requests passed through due to exclusion rules.
+	MetricRequestsExcluded = "requests_excluded"
```

Comment:
Isn't this metric redundant as Request - Disrupted = Excluded?

Reply:
There could be some edge cases when this is not true. For example, this internal error in the GRPC proxy would break that assumption:

https://github.com/grafana/xk6-disruptor/blob/proxy-metrics/pkg/agent/protocol/grpc/handler.go#L58-L64

I think it's better to be explicit, so if the numbers don't match, we can tell something is odd (either a bug in the code counting the metrics, or unexpected errors ocurring).